tardis/io/util.py

```diff
@@ -65,7 +66,18 @@ class YAMLLoader(yaml.Loader):
     A custom YAML loader containing all the constructors required
     to properly parse the tardis configuration.
     """
-
+    def demo_constructor(self, node):
+    	data = self.construct_scalar(node)
+    	return data
+    def spectrum_constructor(self, node):
+    	"""
+    	A constructor to convert spectrum values to nparray
+    	"""
+    	data = self.construct_scalar(node)
+    	data = data.split('%')
+        start = u.Quantity(data[0], data[3])
+        stop = u.Quantity(data[1], data[3])
+        return data
```

Comment:
- Indentation seems wrong
- `spectrum_constructor` is way too restrictive for a name that constructs ranges in general.
- You create start and stop local variables without using them again.
- Your delimiting scheme seems different from that specified in #687.
- You seem to ignore the `num` property, you don't handle cases where the unit is omitted/not required.
- You return a list and not a linspace.
- The `spectrum_constructor` doesn't seem to be used at all in the end.